sfge/src/main/java/com/salesforce/logging/LoggerUtil.java

```diff
@@ -0,0 +1,51 @@
+package com.salesforce.logging;
+
+import com.google.gson.Gson;
+import com.salesforce.graph.vertex.BaseSFVertex;
+import com.salesforce.messaging.CliMessager;
+import com.salesforce.messaging.EventKey;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.Logger;
+
+public final class LoggerUtil {
+
+    public static void warnAndPostTelemetry(Logger logger, String msg, BaseSFVertex vertex) {
```

Comment:
If the vertex is unavailable, we can have an overload of this method that doesn't accept a vertex, and the vertex property would be "UNAVAILABLE" instead.

Reply:
We might have an ApexValue instead of a vertex in some cases.